### ISSUE TITLE
Is this still working with Atom 1.0.15?

### ISSUE THREAD

**Issue by guyellis** [peterdotjs/atom-synced-sidebar#13]
First off: Thank you! This is an awesome plugin! Great productivity! I've been using it for around 6 months. I've noticed that it seems to have stopped working recently. I'm trying to work out if anyone else has it working on Atom 1.0.15? i.e. is the problem me and my configuration or has it stopped working for recent versions of Atom? Thanks.

**Comment by guyellis** [comment]
Let me clarify. It semi-works. If the directory tree is open to the file then it will sync. Otherwise it will select the parent directory but not expand it as it used to.

**Comment by peterdotjs** [comment]
@guyellis I've gotten mixed reports - some folks have gotten it working and one reported it still not working. Haven't really had time to debug the issue - I'll try to get to it soon.  Glad you've found it useful. 

Sounds buggy or broken from your last comment - I'll try to get it up and running again. Thanks for your patience.

**Comment by kevinsimper** [comment]
it does not work for me either

![image](https://cloud.githubusercontent.com/assets/1126497/10068562/a09003dc-62a5-11e5-85fd-b1e8265e4049.png)

**Comment by aknuds1** [comment]
Ah, I'm also having the issue of containing folders not auto-expanding. I guess it used to work?

**Comment by aknuds1** [comment]
Might it be easy to fix? I know my way around JavaScript (wrote a patch for Chromium Dev Tools once f.ex.), but have never hacked on Atom plugins before.

**Comment by aknuds1** [comment]
It doesn't scroll into view either AFAICT.

**Comment by kevinsimper** [comment]
It works for me again, weird!

**Comment by peterdotjs** [comment]
This is getting out of hand - looking into it. Been pretty busy with work, will try to resolve ASAP.

**Comment by peterdotjs** [comment]
I've tested with a clean install of the latest atom v1.0.19 and it is working properly. @guyellis @aknuds1 can you guys check? If it's still buggy it could be a conflicting package.

**Comment by Stanzilla** [comment]
@peterdotjs Still getting this error 

```
> runas@2.0.0 install C:\Users\Benjamin\AppData\Local\Temp\apm-install-dir-115829-6268-l5n9xh\node_modules\synced-sidebar\node_modules\pathwatcher\node_modules\runas
> node-gyp rebuild


C:\Users\Benjamin\AppData\Local\Temp\apm-install-dir-115829-6268-l5n9xh\node_modules\synced-sidebar\node_modules\pathwatcher\node_modules\runas>if not defined npm_config_node_gyp (node "C:\Users\Benjamin\AppData\Local\atom\app-1.0.19\resources\app\apm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild ) 

gypnpm ERR! Windows_NT 6.2.9200
npm ERR! argv "C:\\Users\\Benjamin\\AppData\\Local\\atom\\app-1.0.19\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\Benjamin\\AppData\\Local\\atom\\app-1.0.19\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\Benjamin\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\Benjamin\\.atom\\.apmrc" "install" "C:\\Users\\Benjamin\\AppData\\Local\\Temp\\d-115829-6268-16hsx0\\package.tgz" "--target=0.30.7" "--arch=ia32" "--msvs_version=2013"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! runas@2.0.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the runas@2.0.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the runas package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls runas
npm ERR! There is likely additional logging output above.
```

like I mentioned in my issue.

Here's a list of the packages I have installed:

```
C:\Users\Benjamin\.atom\packages (36)
├── atom-beautify@0.28.14
├── atom-format-lua@0.0.5
├── auto-detect-indentation@0.4.2
├── auto-reveal-in-sidebar@0.5.0
├── autocomplete-paths@1.0.2
├── autocomplete-php@0.3.7
├── autocomplete-sass@0.1.0
├── build@0.47.0
├── build-gulp@0.2.0
├── css-color-underline@1.0.1
├── file-icons@1.6.9
├── file-watcher@0.3.1
├── fold-functions@0.3.1
├── indent-guide-improved@1.4.3
├── indentation-indicator@0.6.0
├── keyboard-localization@1.4.8
├── language-blade@0.16.0
├── language-dots@0.0.4
├── language-gitignore@0.2.0
├── language-lua@0.9.4
├── language-powershell@2.1.0
├── laravel@0.7.0
├── less-than-slash@0.8.0
├── line-ending-selector@0.0.5
├── linter@1.6.0
├── linter-lua@1.0.0
├── linter-lua-findglobals@1.0.0
├── linter-luacheck@1.1.2
├── linter-php@1.0.4
├── minimap@4.13.4
├── open-recent@2.3.0
├── quick-editor@0.4.3
├── revert-buffer@0.5.0
├── svn@0.0.8
├── tabs-to-spaces@0.11.1
└── trailing-spaces@0.3.2
```

**Comment by guyellis** [comment]
@peterdotjs Thanks for the time you're putting into this - much appreciated.

I just upgraded to Atom 1.0.19 and tried again, unfortunately the same behavior: The folder will be selected but it no longer auto-expands.

I also found an old Ubuntu VM that I hadn't touched since June and opened that. It has Atom 0.218.0 in it and it already had synced-sidebar installed. I test it on that and on 0.218.0 it worked. I then upgraded Atom on the VM to 1.0.19 and it stopped working again with the same symptoms: i.e. selects parent folder but won't expand it.

I also tried disabling all community packages except synced-sidebar to see if it was a conflict in any of those but that didn't help. Same problem.

All of the above was tried on Atom running on Ubuntu 14.04.

**Comment by Stanzilla** [comment]
Maybe some packages you depend on don't run on the new Node version Atom comes with?

**Comment by aknuds1** [comment]
Thanks for looking into this @peterdotjs :) It appears to be working in 1.0.19. Containing folders are auto-expanded and the file is scrolled into view.

**Comment by Stanzilla** [comment]
So the screenshot @kevinsimper posted mentions pathwatcher (https://github.com/atom/node-pathwatcher) and runas (https://github.com/atom/node-runas), you don't use runas directly but pathwatcher does (https://github.com/atom/node-pathwatcher/blob/master/package.json). It requires version "^3.1.0" of it.

Looking at the commits to pathwatcher (https://github.com/atom/node-pathwatcher/commits/master) tells us there was some toying with compilation on older Node versions and the update to NaN 2 (https://github.com/atom/node-pathwatcher/commit/7edcf74694281ec79b0229785ae4312d02ec0455) which is backwards _incompatible_.

Your package depends on pathwatcher "^3.0.0" (https://github.com/peterdotjs/atom-synced-sidebar/blob/master/package.json#L19) which is pre-NaN 2 while pathwatcher is already at 6.2.4 (https://github.com/atom/node-pathwatcher/releases/tag/v6.2.4) and comes with NaN 2.0.

Why it starts working again for people that have already installed is probably because Atom learned to rebuild packages and can probably fix incompatibilities for you when a package is already installed?

tl;dr: Try updating every dependency you have to a version that only uses NaN 2.0+ and it should work again.

**Comment by Stanzilla** [comment]
@peterdotjs any news here? :)

**Comment by yoyurec** [comment]
+1

**Comment by peterdotjs** [comment]
@Stanzilla I've updated v 0.4.1 with newer versions of dependencies. Hoping this resolves any of the version incompatibility issues.

**Comment by Stanzilla** [comment]
@peterdotjs Hrm, sadly not. Now I really don't know what else to try :( Thanks for trying though!

```
> runas@3.1.0 install C:\Users\Benjamin\AppData\Local\Temp\apm-install-dir-11598-11684-etg5y1\node_modules\synced-sidebar\node_modules\pathwatcher\node_modules\runas
> node-gyp rebuild


C:\Users\Benjamin\AppData\Local\Temp\apm-install-dir-11598-11684-etg5y1\node_modules\synced-sidebar\node_modules\pathwatcher\node_modules\runas>if not defined npm_config_node_gyp (node "C:\Users\Benjamin\AppData\Local\atom\app-1.0.19\resources\app\apm\node_modules\npm\bin\node-gyp-bin\\..\..\node_modules\node-gyp\bin\node-gyp.js" rebuild )  else (node  rebuild ) 

gypnpm ERR! Windows_NT 6.2.9200
npm ERR! argv "C:\\Users\\Benjamin\\AppData\\Local\\atom\\app-1.0.19\\resources\\app\\apm\\bin\\node.exe" "C:\\Users\\Benjamin\\AppData\\Local\\atom\\app-1.0.19\\resources\\app\\apm\\node_modules\\npm\\bin\\npm-cli.js" "--globalconfig" "C:\\Users\\Benjamin\\.atom\\.apm\\.apmrc" "--userconfig" "C:\\Users\\Benjamin\\.atom\\.apmrc" "install" "C:\\Users\\Benjamin\\AppData\\Local\\Temp\\d-11598-11684-y2k58g\\package.tgz" "--target=0.30.7" "--arch=ia32" "--msvs_version=2013"
npm ERR! node v0.10.40
npm ERR! npm  v2.13.3
npm ERR! code ELIFECYCLE

npm ERR! runas@3.1.0 install: `node-gyp rebuild`
npm ERR! Exit status 1
npm ERR! 
npm ERR! Failed at the runas@3.1.0 install script 'node-gyp rebuild'.
npm ERR! This is most likely a problem with the runas package,
npm ERR! not with npm itself.
npm ERR! Tell the author that this fails on your system:
npm ERR!     node-gyp rebuild
npm ERR! You can get their info via:
npm ERR!     npm owner ls runas
npm ERR! There is likely additional logging output above.
```

**Comment by guyellis** [comment]
Likewise - didn't solve the folder not opening problem that I'm seeing. Thanks for trying @peterdotjs - appreciate that!

**Comment by steelbrain** [comment]
Guys just so you would know that all the dependencies required by this package are redundant. so here's what you can do

```
cd ~/.atom/packages
git clone https://github.com/peterdotjs/atom-synced-sidebar synced-sidebar --depth=1
```

and start atom

**Comment by guyellis** [comment]
@steelbrain Good idea - unfortunately still no cigar. Thanks!

**Comment by steelbrain** [comment]
btw, run atom with `--dev`, it'll report the API deprecations then

**Comment by peterdotjs** [comment]
:( @Stanzilla, I'll try to repo your setup with the package list you listed earlier. 

@steelbrain there aren't any API deprecations - that I can be certain of. 

@guyellis can you list the packages that you have? It sounds like the issue you are having is a conflicting package.

**Comment by steelbrain** [comment]
@peterdotjs just a side question: Why is there a `npm-debug.log` in the repo?

**Comment by peterdotjs** [comment]
@steelbrain shouldn't be :( I'll remove it update `.gitignore`. Thanks for the catch!

**Comment by Stanzilla** [comment]
I mean the package works fine, just the install process somehow does not want to. I installed it manually now and that works as well, so it must be something during install via apm what breaks it.

**Comment by steelbrain** [comment]
@Stanzilla I repeat, **all** the dependencies referenced by this package are redundant. What's failing is the `pathwatcher` I think because it has native dependencies and since the upgrades in electron recently those native APIs have been broken and this package hasn't bumped the number yet.

If you edit this package's `package.json` and completely drop the dependencies section and never install them, it would still work :)

**Comment by Stanzilla** [comment]
@steelbrain Ah, I see. I thought he updated/fixed all the dependencies already. My bad!

**Comment by steelbrain** [comment]
@peterdotjs Why did you update them and not remove them? :smiley:

**Comment by peterdotjs** [comment]
@steelbrain - from a previous comment by @Stanzilla I tried to update the version of `pathwatcher` - there was no plan to remove them. When I started the package I blindly took the initial dependencies from [atom tree view](https://github.com/atom/tree-view/blob/master/package.json) since the package was very similar to what I wanted to achieve. 

`atom-space-pen-views` and `minimatch` aren't listed as part of the atom `package.json` so I'll confirm to see if these can be removed as well.

**Comment by steelbrain** [comment]
@peterdotjs there's only one `require` call in https://github.com/peterdotjs/atom-synced-sidebar/blob/master/lib/synced-sidebar.coffee and that's to atom. so Yeah, you can drop all of them.
Also note that when you manipulate `tree-view` you **do not** need to re-list it's dependencies. Only list the ones that you use in your own project.

**Comment by peterdotjs** [comment]
@steelbrain yeah I got that - I had used that as a skeleton and never cleaned it up.

**Comment by peterdotjs** [comment]
`v0.4.2` has removed the unused dependencies. @Stanzilla @guyellis  - heres to hoping this resolves your install issue.

**Comment by guyellis** [comment]
@peterdotjs That did it! It now expands the folder and highlights the file. You are awesome!

I thought that @steelbrain suggestion was the same as what you did? I'm guessing not.

**Comment by steelbrain** [comment]
@guyellis It is.

**Comment by guyellis** [comment]
@steelbrain Well that's really weird. Id a `rm -rf <sync-sidebar-etc.>` before running that command and then started atom.

**Comment by Stanzilla** [comment]
@peterdotjs all good now :+1:
